src/coreclr/nativeaot/System.Private.CoreLib/src/System/InvokeUtils.cs

```diff
@@ -134,87 +134,341 @@ private static Exception ConvertOrWidenPrimitivesEnumsAndPointersIfPossible(obje
                 return CreateChangeTypeArgumentException(srcEEType, dstEEType);
             }
 
+            EETypeElementType srcElementType = srcEEType->ElementType;
+
+            void* rawDstValue = null;
+
+            Unsafe.SkipInit(out char charValue);
+            Unsafe.SkipInit(out sbyte sbyteValue);
+            Unsafe.SkipInit(out byte byteValue);
+            Unsafe.SkipInit(out short shortValue);
+            Unsafe.SkipInit(out ushort ushortValue);
+            Unsafe.SkipInit(out int intValue);
+            Unsafe.SkipInit(out uint uintValue);
+            Unsafe.SkipInit(out long longValue);
+            Unsafe.SkipInit(out ulong ulongValue);
+            Unsafe.SkipInit(out float floatValue);
+            Unsafe.SkipInit(out double doubleValue);
+
+            Unsafe.SkipInit(out dstObject);
+
+            // This switch supports the conversions present in the table below, in 'CanPrimitiveWiden'
             switch (dstElementType)
             {
                 case EETypeElementType.Boolean:
-                    dstObject = Convert.ToBoolean(srcObject);
+
+                    // Can only be bool here
+                    dstObject = srcObject;
                     break;
 
                 case EETypeElementType.Char:
-                    char charValue = Convert.ToChar(srcObject);
-                    dstObject = dstEEType->IsEnum ? Enum.ToObject(dstEEType, charValue) : charValue;
+                    switch (srcElementType)
+                    {
+                        case EETypeElementType.Char:
+                            charValue = RuntimeHelpers.FastUnbox<char>(srcObject);
```

Comment:
```suggestion
                            charValue = Unsafe.As<byte, char>(ref rawSrcValue);
```
I would rather call Unsafe.As directly. I am not fan of non-local "Fast" helpers with special behaviors. Makes it hard it tell what's going on.``

Reply:
Fixed in 84fbc82eca09f3328327ac12b5067a744e80540d 🙂